apps/api/src/domain/models/__tests__/buildResourceModelContextStateValidatorTestCases/buildSongResourceModelContextStateValidatorTestCase.ts

```diff
@@ -0,0 +1,51 @@
+import InconsistentTimeRangeError from '../../../domainModelValidators/errors/context/invalidContextStateErrors/timeRangeContext/InconsistentTimeRangeError';
+import { resourceTypes } from '../../../types/resourceTypes';
+import {
+    TimeRangeContext,
+    TimeRangeWithoutData,
+} from '../../context/time-range-context/time-range-context.entity';
+import { EdgeConnectionContextType } from '../../context/types/EdgeConnectionContextType';
+import buildAllInvalidTestCasesForResource from '../utilities/buildAllInvalidTestCasesForResource';
+import buildAllValidTestCasesForResource from '../utilities/buildAllValidTestCasesForResource';
+
+const validCases = buildAllValidTestCasesForResource(resourceTypes.song);
+const inconsistentContextTypeTestCases = buildAllInvalidTestCasesForResource(resourceTypes.song);
```

Comment:
Just curious why this is `inconsistent` rather than `invalid`?

Reply:
This helper `buildAllInvalidTestCasesForResource` is poorly named. What it actually does is to build all test cases where the `contextType` of the context model is inconsistent with the `allowedContextTypes` for the resource model. It doesn't build any cases where the context type is actually allowed but the state is inconsistent with the resource model.

Yet another reminer that we need to rename the helper to `buildAllInconsistentContextTypeTestCases`.

Reply:
renamed the helper